tests/test_group_by.py

```diff
@@ -1,14 +1,19 @@
 from __future__ import annotations
 
+from typing import Any
+
 import pandas as pd
 import polars as pl
 import pytest
 
 import narwhals as nw
 from tests.utils import compare_dicts
 
-df_pandas = pd.DataFrame({"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]})
-df_lazy = pl.LazyFrame({"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]})
+data_1 = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]}
+data_2 = {"a": [1, 1, 3], "b": [4, 4, 6], "c": [7, 8, 9]}
```

Comment:
any better names for these? 🤔

`data_2` is the data used for the tests that were in `group_by_test.py`

Reply:
Can't we just keep `data_1` for both tests and change `expected` of `test_group_by_iter` accordingly? Am I missing something?

Reply:
Yes I think it makes sense!
I am now using the previous `data_2` for all. It had repeating values in 2 columns, I think that is why `test_group_by_iter`  was using it. What do you think?

Reply:
Oh I see, thanks for pointing that out! Anyway, keeping the one dataset that works for both tests is all we need! Thanks for the adjustment